internal/handlers/update.go

```diff
@@ -253,3 +267,17 @@ func (sh *ServerHandler) getResponseMetric(metric entities.Metrics) (*entities.M
 		return &currentDelta, nil
 	}
 }
+
+func (sh *ServerHandler) isBodyValid(data []byte, reqHash string) bool {
```

Comment:
You should check the body integrity in a middleware

Reply:
I thought of doing this, but once I read the body in a middleware I need to implement a custom handler to write back the body and reread it in the uploads handler. I thought it would be ok to keep it this way as it's only used in one handler.

Reply:
Generally speaking, it is not too complicated to implement and the overhead of this operation should be negligible: you should limit the length of the body anyway, and the IO-bound operations would take the majority of time. Here's an approach you can use: https://stackoverflow.com/questions/46948050/how-to-read-request-body-twice-in-golang-middleware

However, you can of course do it like this for now, as there is no code to reuse (as you've mentioned). So you can resolve this comment

Reply:
Yeah, it doesn't seem too complicated. I bookmarked the link and I will keep it for the future. Although, I would leave it as is for, need to get this PR approved ASAP.